src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -483,16 +484,34 @@ private static bool IsNamedTypeAccessible(NamedTypeSymbol type, Symbol within, r
             Debug.Assert(type.IsDefinition);
             Debug.Assert(baseType.IsDefinition);
 
+            PooledHashSet<NamedTypeSymbol> interfacesLookedAt = null;
+            ArrayBuilder<NamedTypeSymbol> baseInterfaces = null;
+
+            bool baseTypeIsInterface = baseType.IsInterface;
+            if (baseTypeIsInterface)
+            {
+                interfacesLookedAt = PooledHashSet<NamedTypeSymbol>.GetInstance();
+                baseInterfaces = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            }
+
             PooledHashSet<NamedTypeSymbol> visited = null;
             var current = type;
             bool result = false;
 
             while ((object)current != null)
             {
-                if (current.Equals(baseType))
+                if (baseTypeIsInterface == current.IsInterfaceType())
                 {
-                    result = true;
-                    break;
+                    if (current.Equals(baseType))
```

Comment:
>if [](start = 20, length = 2)

Nested if statements could be a single if statement, please.

Reply:
> Nested if statements could be a single if statement, please.

Adjusted

---
In reply to: [264411257](https://github.com/dotnet/roslyn/pull/34012#discussion_r264411257) [](ancestors = 264411257)